tests/datawidgetstests/test_autovizwidget.py

```diff
@@ -97,4 +102,5 @@ def test_convert_to_displayable_dataframe():
     assert_equals(result.dtypes[u'bool_col'], object)
     assert_equals(len(result[u'bool_col']), 2)
     assert_equals(result[u'bool_col'][0], 'True')
-    assert_equals(result[u'bool_col'][1], 'False')
\ No newline at end of file
+    assert_equals(result[u'bool_col'][1], 'False')
+    spark_events.emit_graph_render_event.assert_called_once_with(encoding.chart_type)
```

Comment:
Cool. However, this assertion is part of `test_convert_to_displayable_dataframe`, and it should be part of `test_on_render_viz`. In that test, you could simulate changing the encoding and verifying that the emission changes chart types as the encoding changes.